modules/4_nodes/nodes.tf

```diff
@@ -191,6 +202,18 @@ resource "openstack_compute_instance_v2" "worker" {
     }
 }
 
+resource "openstack_blockstorage_volume_v2" "worker" {
+  count       = var.worker["data_volume_count"] * var.worker["count"]
+  name        = "${var.cluster_id}-worker-${count.index}-volume"
+  size        = var.worker["data_volume_size"]
+}
+
+resource "openstack_compute_volume_attach_v2" "worker" {
+  count        = var.worker["data_volume_count"] * var.worker["count"]
+  instance_id  = openstack_compute_instance_v2.worker.*.id[floor(count.index / var.worker["data_volume_count"])]
```

Comment:
Would not it give divide by zero error using the default value? Please use locals with lookup() function.

Reply:
> Would not it give divide by zero error using the default value?

It would not as the resource count will be 0. But still I recommend to use lookup value everywhere as I think the variables can be optional.